cmd/influxd/launcher/launcher.go

```diff
@@ -970,6 +975,40 @@ func (m *Launcher) initTracing(opts *InfluxdOpts) {
 	}
 }
 
+// writePIDFile will write the process ID to pidFile and register a cleanup function to delete it during
+// shutdown. If pidFile is empty, then no PID file is written and no cleanup function is registered.
+func (m *Launcher) writePIDFile(pidFile string) error {
+	if pidFile == "" {
+		return nil
+	}
+
+	// Create directory to PIDfile if needed.
+	if err := os.MkdirAll(filepath.Dir(pidFile), 0777); err != nil {
+		return fmt.Errorf("mkdir: %w", err)
+	}
+
+	// Write PID to file
+	pidStr := strconv.Itoa(os.Getpid())
+	if writeErr := os.WriteFile(pidFile, []byte(pidStr), 0666); writeErr != nil {
+		// Let's make sure we don't leave a PID file behind on error.
+		removeErr := os.Remove(pidFile)
+		return fmt.Errorf("write file: %w; remove file: %w", writeErr, removeErr)
```

Comment:
I'm wondering if we want to use something lower level than `WriteFile` like `OpenFile` so we can pass in `O_EXCL` and detect existing PID files that weren't cleaned up (or multiple InfluxDB instances).  I think that might be an error we want to detect, report, and possibly abort on.

Reply:
I tend to find using the PID file as a lock file more annoying than useful (looking at you PostgreSQL!). It also generates [tons of support questions](https://www.google.com/search?q=postgresql+error+pid+lock+file).

The less annoying way of preventing multiple instances is to check for for already listening sockets. We kind of do that today by aborting if we can't bind to the bind addresses. We do the abort after we've opened the PID file, so we would blow away the PID file on shutdown in those situations even though the other (presumed) influxd instance is running.

If we want more robust handling of the PID file if you accidentally launch multiple influxd instances with the same configuration, my suggestion would be a feature request to try connecting to the bind addresses very early on to see if anything is listening on them, or use a Unix domain socket / named pipe for the check. This way when the application dies, there is nothing listening or responding on the socket. With a PID or lock file, the file can be stay around if the application is not shut down cleanly.

Or we could simply write the PID _after_ we bind the sockets.

Reply:
On second thought, I don't like the idea of dropping the PID file _after_ we're listening for connections, because that could put a long delay between `influxd` starts and when the PID file drops.

Reply:
My suggestion then, to avoid support questions, would be to check for existence, issue a warning if present, and then continue, rather than a fatal error.  That will let the customer know that the WAL flush may not have happened on shutdown, for instance.

Reply:
Maybe we can do a compromise between the PID approaches. If you specify `--pid-file`, the default behavior is to abort with an error if the PID file already exists and the PID file is not touched. If you additionally specify a binary flag `--overwrite-pid-file`, then a warning is issued if the PID file is already present, but it will be overwritten and startup will continue.

Adding an extra parameter does add complexity, but the impact of `--overwrite-pid-file` would be extremely localized and simple to exhaustively test in the automated unit tests.

Thoughts?